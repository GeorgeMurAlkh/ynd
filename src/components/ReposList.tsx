import { IRepo } from '../models/repo.model';

interface ReposListProps {
  repos: IRepo[] | undefined
}

const ReposList = ({ repos }: ReposListProps) => {
  return (
    <>
      <div className="card m-2">
        <div className="list-group bg-primary">
        {repos && repos.map((repo, i) => {
            return (
              <a className={`list-group-item list-group-item-action text-break ${i < repos.length - 1 && 'border-bottom border-secondary'}`}
              href={repo.html_url} target="_blank" 
              rel="noreferrer"
              key={repo.id}
              data-testid={`item-${i}`}
              >
                <div className="d-flex align-items-center justify-content-between text-start">
                  <div className="align-items-start pe-2">
                    <h5>
                    {repo.name}
                    </h5>
                    <p>
                      {repo.description}
                    </p>          
                  </div>
                  <div className="d-flex">
                    <i className="bi bi-star-fill me-1" />
                    <span className="unbrake">{repo.stargazers_count}</span>
                  </div>
                </div>
              </a>
            );
          }
        )}
        </div>
      </div>
    </>
  );
};

export default ReposList;
