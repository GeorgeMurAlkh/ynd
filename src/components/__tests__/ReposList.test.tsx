import { render, screen, cleanup } from '@testing-library/react';
import ReposList from '../ReposList';
import renderer from 'react-test-renderer';
import { IRepo } from '../../models/repo.model';

const repos: IRepo[] = [
  {
    id: 1,
    name: 'somelogin',
    stargazers_count: 434,
    description: 'some desc',
    html_url: 'idefinatelyshouldmockthisdata:)'
  }
];

afterEach(() => {
  cleanup();
});

test('should render todo component', () => {
  render(<ReposList repos={repos}></ReposList>);
  const element = screen.getByTestId('item-0');
  expect(element).toBeInTheDocument();
});

test('matches snapshot', () => {
  const tree = renderer.create(<ReposList repos={repos}></ReposList>);
  expect(tree).toMatchSnapshot();
});