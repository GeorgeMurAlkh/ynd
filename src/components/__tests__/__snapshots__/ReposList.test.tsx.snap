// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card m-2"
>
  <div
    className="list-group bg-primary"
  >
    <a
      className="list-group-item list-group-item-action text-break false"
      data-testid="item-0"
      href="idefinatelyshouldmockthisdata:)"
      rel="noreferrer"
      target="_blank"
    >
      <div
        className="d-flex align-items-center justify-content-between text-start"
      >
        <div
          className="align-items-start pe-2"
        >
          <h5>
            somelogin
          </h5>
          <p>
            some desc
          </p>
        </div>
        <div
          className="d-flex"
        >
          <i
            className="bi bi-star-fill me-1"
          />
          <span
            className="unbrake"
          >
            434
          </span>
        </div>
      </div>
    </a>
    ;
  </div>
</div>
`;
